topo/node/srl/srl_test.go

```diff
@@ -179,7 +177,10 @@ func TestGenerateSelfSigned(t *testing.T) {
 			testFile: "generate_certificate_failure",
 		},
 	}
-	logging.SetDebugLogger(log.Print)
+
+	scraplilogging.WithLogger(log.Print)
+	scraplilogging.WithLevel(scraplilogging.Debug)
+
```

Comment:
@carlmontanari 👋 
Is this the right alternative to the prev func?

Reply:
not quite. logging is now per connection (which imo makes way more sense... why it was never like that.... I dunno), so there is no global setup like that. you've got two options:

1) you can just use the `WithDefaultLogger` option func (passing it in as an option during driver creation) like:

```go
	d, err := generic.NewDriver(
		// whatever stuff you need of course
		// the WithDefaultLogger option applies a simple log.Print logger at level info
		options.WithDefaultLogger(),
	)
```

if you wanted to change the level to debug when doing it like this you could do `d.Logger.Level = "debug"` (kinda janky, but works)

2) alternatively you can create a `logging.Instance` (probably w/ `NewInstance`) and set whatever logger(s) you want and at whatever level you want with those `WithLogger` and `WithLevel` options you have in this section. That'd look like:

```go
func myLoggerFunc(x ...interface{}) {
	// do some logging whatever
}

func main() {
	li, err := logging.NewInstance(
		logging.WithLevel(logging.Debug),
		logging.WithLogger(log.Print),
		logging.WithLogger(myLoggerFunc),
	)
	if err != nil {
		fmt.Printf("failed to logging instance; error: %+v\n", err)

		return
	}

	d, err := generic.NewDriver(
		// whatever stuff you need of course
		options.WithLogger(li),
	)
```

Reply:
roger that. I think opt1 can't be used in my case, since from the test package I don't have access to the driver struct, it gets created in a non test part of the code. All I pass over is options.
So likely if I need to set a debug level, I just need to use option 2